wallet/handler.go

```diff
@@ -203,21 +198,38 @@ func (h *Handler) SignTxV2(token string, req walletpb.SubmitTransactionRequest,
 	return commands.NewTransaction(keyPair.pubBytes, marshalledData, signature), nil
 }
 
-func (h *Handler) TaintKey(token, pubKey, passphrase string) error {
-	h.mu.Lock()
-	defer h.mu.Unlock()
+func (h *Handler) VerifyAny(name, inputData, sig, pubKey string) (bool, error) {
+	h.mu.RLock()
+	defer h.mu.RUnlock()
 
-	name, err := h.auth.VerifyToken(token)
+	rawSig, err := base64.StdEncoding.DecodeString(sig)
 	if err != nil {
-		return err
+		return false, err
+	}
+
+	rawInputData, err := base64.StdEncoding.DecodeString(inputData)
+	if err != nil {
+		return false, err
+	}
+
+	kp, err := h.getKeyPair(name, pubKey)
+	if err != nil {
+		return false, err
 	}
```

Comment:
Not sure what does getKeyPair but surely you should not be able to get a kepair just from a pubkey right?
Also verifying a signature should required only the pubkey + the signature and the data to sign I suppose. We could also expose that as an endpoint for free

Reply:
You are right. I fixed this. About the endpoint, I'll make another PR